src/Gui/TaskView/TaskImageScale.cpp

```diff
@@ -85,4 +112,196 @@ void TaskImageScale::changeHeight()
     }
 }
 
+View3DInventorViewer* TaskImageScale::getViewer() const
+{
+    if (!feature.expired()) {
+        auto vp = Application::Instance->getViewProvider(feature.get());
+        auto doc = static_cast<ViewProviderDocumentObject*>(vp)->getDocument();
+        auto view = dynamic_cast<View3DInventor*>(doc->getViewOfViewProvider(vp));
+        if (view) {
+            return view->getViewer();
+        }
+    }
+
+    return nullptr;
+}
+
+void TaskImageScale::selectedPoints(size_t num)
+{
+    if (num == 1) {
+        ui->labelInstruction->setText(tr("Select second point"));
+    }
+    else if (num == 2) {
+        ui->labelInstruction->setText(tr("Enter distance value"));
```

Comment:
Maybe:
```suggestion
        ui->labelInstruction->setText(tr("Enter physical distance between the two selected points"));
```

Reply:
Haven't checked this so I don't know how crowded the UI is, if a more compact text than chennes' proposal is desired consider:
```suggestion
        ui->labelInstruction->setText(tr("Enter desired distance between the points"));
```

Reply:
Shorter and cleaner, `Enter the distance between the points` (the points were selected before by the user)

Reply:
> Shorter and cleaner, Enter the distance between the points

The value that is set to the spin box after the second click is already the physical distance. The wording must be clear that the **desired** distance is meant. The ratio of desired and current distance gives the scaling factor.